drivers/sensor/wsen_hids/hids.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (c) 2022 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_hids
+
+#include <string.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+#include "hids.h"
+
+LOG_MODULE_REGISTER(HIDS, CONFIG_SENSOR_LOG_LEVEL);
+
+struct str2odr {
+	const char *str;
+	HIDS_outputDataRate_t odr;
+};
+
+/* Maps ODR configuration string (CONFIG_HIDS_ODR) to enum value used by HIDS driver. */
+static const struct str2odr hids_odr_string_list[] = {
+	{ "1", HIDS_odr1Hz },
+	{ "7", HIDS_odr7Hz },
+	{ "12.5", HIDS_odr12_5Hz },
+};
+
+/*
+ * List of supported output data rates (sensor_value struct, input to
+ * sensor_attr_set()). Index into this list is used as argument for
+ * HIDS_setOutputDataRate().
+ */
+static const struct sensor_value hids_odr_list[] = { { .val1 = 0, .val2 = 0 },
+						     { .val1 = 1, .val2 = 0 },
+						     { .val1 = 7, .val2 = 0 },
+						     { .val1 = 12, .val2 = 5 * 100000 } };
```

Comment:
clang might be able to format this if you have comma

```suggestion
{ .val1 = 12, .val2 = 5 * 100000 }, };
```

otherwise it'd be nice if you can spend some time to format this manually

Reply:
Done.